stock_account_unfuck/models/stock.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import division
+
+from openerp import api, models
+
+
+class StockMove(models.Model):
+
+    _inherit = "stock.move"
+
+    @api.multi
+    def _recalculation_average_cost_price(self, signal=1):
+        """ move is a browse record """
+        for move in self:
+            if move.product_qty == 0:
+                return
+            for quant in move.reserved_quant_ids:
+                if quant.qty <= 0:
+                    return
+            average_valuation_price = 0.0
+            qty_available = move.product_id.product_tmpl_id.qty_available
+            average_valuation_price = sum(
+                [quant.qty * quant.cost for quant in move.reserved_quant_ids])
```

Comment:
This is right when returning merchandise to Suppliers.
```
+            average_valuation_price = sum(
+                [quant.qty * quant.cost for quant in move.reserved_quant_ids])
```

When returning merchandise from `customer` or `transit`, the value to be used is the 
average price when the transaction was performed, i.e., `price_unit ` on
stock.move record.
https://github.com/odoo/odoo/blob/8.0/addons/stock/stock.py#L1811